clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/KafkaStoreIngestionService.java

```diff
@@ -1054,6 +1073,76 @@ public AdminResponse getConsumptionSnapshots(String topicName, ComplementSet<Int
     return response;
   }
 
+  /**
+   * Return the metadata information for the given store. The data is retrieved from its respective repositories which
+   * originate from the VeniceServer.
+   * @param storeName
+   * @return {@link MetadataResponse} object that holds all the information required for answering a server metadata
+   * fetch request.
+   */
+  @Override
+  public MetadataResponse getMetadata(String storeName) {
+    MetadataResponse response = new MetadataResponse();
+    try {
+      Store store = metadataRepo.getStoreOrThrow(storeName);
+
+      List<Integer> versions = new ArrayList<>();
+      for (Version version: store.getVersions()) {
+        version.getPartitionerConfig();
+        versions.add(version.getNumber());
+      }
+      Map<CharSequence, CharSequence> partitionerParams =
+          new HashMap<>(store.getPartitionerConfig().getPartitionerParams());
+      VersionProperties versionProperties = new VersionProperties(
+          store.getCurrentVersion(),
+          versions,
+          store.getCompressionStrategy().getValue(),
```

Comment:
Let's use the values from the actual version. e.g. `store.getVersion(store.getCurrentVersion()).get().getCompressionStrategy();` Since the version configs are immutable but store configs can be updated and could go out of sync. i.e. the store level compression config was just updated and actually doesn't match with any of the existing versions.

Reply:
Does this also apply to `partitionCount` and `partitionerConfig`?